python/ray/data/dataset.py

```diff
@@ -1007,6 +1007,11 @@ def sum(self, on: Union[KeyFn, List[KeyFn]] = None) -> U:
                 - For an Arrow dataset: it can be a column name or a list
                   thereof, and the default is to return an ``ArrowRow``
                   containing the column-wise sum of all columns.
+            ignore_nulls: Whether to ignore null values. If ``True``, null
+                values will be ignored when computing the sum; if ``False``,
+                if a null value is encountered, the output will be null.
```

Comment:
nit: the output will be None

Reply:
Yep good catch, I'll update the documentation of the output.